app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ParkingView.java

```diff
@@ -0,0 +1,26 @@
+package com.onboarding.parkingsystemjava.mvp.view;
+
+import android.app.Activity;
+import android.widget.Toast;
+
+import com.onboarding.parkingsystemjava.R;
+import com.onboarding.parkingsystemjava.databinding.ActivityMainBinding;
+import com.onboarding.parkingsystemjava.mvp.contract.MainActivityContract;
+import com.onboarding.parkingsystemjava.mvp.view.base.ActivityView;
+
+public class ParkingView extends ActivityView implements MainActivityContract.MainActivityView {
+
+    private final ActivityMainBinding binding;
+
+    public ParkingView(Activity activity, ActivityMainBinding binding) {
+        super(activity);
+        this.binding = binding;
+    }
+
+    @Override
+    public void showParkingLotsAvailable(int parkingLots) {
+        if (getContext() != null) {
```

Comment:
you are getting the context 3 times here.. the best approach is to create a context variable, make the validations over the variable and use the variable

Reply:
done, made a local variable in the function and used that variable instead of getContext().